client/src/app/components/input-geo/input-geo.component.ts

```diff
@@ -0,0 +1,197 @@
+import { 
+  Component, OnInit, Input, Output, EventEmitter, HostBinding,
+  Optional, Self, ViewChild, ElementRef
+} from '@angular/core';
+import { FormGroup, NgControl, FormBuilder } from '@angular/forms';
+import { MatFormFieldControl } from '@angular/material/form-field';
+import { Subject } from "rxjs";
+import { coerceBooleanProperty } from '@angular/cdk/coercion';
+import { HttpClient } from '@angular/common/http';
+
+import { environment } from 'src/environments/environment';
+import { config } from './config';
+import { get as getScript } from 'scriptjs';
+import { User } from 'src/app/models/user.model';
+import { AuthService } from 'src/app/services/auth.service';
+declare const google: any;
+
+@Component({
+  selector: 'app-input-geo',
+  templateUrl: './input-geo.component.html',
+  styleUrls: ['./input-geo.component.scss'],
+  providers: [{provide: MatFormFieldControl, useExisting: InputGeoComponent}]
+})
+export class InputGeoComponent implements OnInit, MatFormFieldControl<string> {
+  @Input() dataPoint: string|undefined;
+  @Input() value: string;
+  @ViewChild('geoInput') geoInput: ElementRef;
+
+  targetLocationValue: string|number|undefined = '';
+  @Output() targetLocationChange = new EventEmitter<string>();
+
+  geoForm: FormGroup;
+
+  // START: Implementing the MatFormFieldControl interface
+  controlType = 'input-geo';
+
+  static nextId = 0;
+  @HostBinding() id = `${this.controlType}-${InputGeoComponent.nextId++}`;
+  stateChanges = new Subject<void>();
```

Comment:
Who is subscribed to that?

Reply:
This is a part of the implementing the "MatFormFieldControl": [https://material.angular.io/guide/creating-a-custom-form-field-control#statechanges]